scripts/removeUnusedResourcesFromAssets.swift

```diff
@@ -0,0 +1,104 @@
+                let fileAttributes = try fileURL.resourceValues(forKeys: [.isRegularFileKey])
+                let filePath = fileURL.relativePath
+
+                if fileAttributes.isRegularFile! &&
+                    filePath.hasSuffix(resourceFileSuffix) {
+
+                    print("\nUsed resources in file: \(filePath)")
+
+                    do {
+                        let resourceFileListContents = try String(contentsOf: fileURL)
+
+                        for entry in resourceFileListContents.split(separator: "\n") {
+                            let resourceFileEntry = entry.trimmingCharacters(in: .whitespacesAndNewlines)
+                            usedResources.insert(resourceFileEntry)
+                            print("- \(resourceFileEntry)")
+                        }
+                    } catch {
+                        fatalError("Failed to read contents resource file: \(filePath) \nError: \(error)")
+                   }
+                }
+            } catch {
+                fatalError("Failed to retrieve resource file: \(fileURL) \nError: \(error)")
+            }
+        }
+    }
+
+    return usedResources
+}
+
+/// Iterates through all folders in a given .xcassets file and removes the .colorset or .imageset folders that are not contained in the used resources set.
+/// - Parameters:
+///   - xcassetsPath: Root path of the .xcassets file.
+///   - usedResourcesSet: Set containing the list of paths (relative to the .xcassets root) of resources that should not be removed.
+func removeResources(from xcassetsPath: String, notContainedIn usedResourcesSet: Set<String>) {
+    let resourceExtensions = ["colorset", "imageset"]
+    let xcassetsURL = URL(fileURLWithPath: xcassetsPath)
+    let xcassetsRelativePath = xcassetsURL.relativePath
+
+    print("\n\nProcessing resources of xcassets in path: \(xcassetsPath)")
+    if let directoriesEnumerator = FileManager.default.enumerator(at: xcassetsURL,
+                                                                  includingPropertiesForKeys: [.isDirectoryKey],
+                                                                  options: [.skipsHiddenFiles, .skipsPackageDescendants]) {
+        for case let directoryURL as URL in directoriesEnumerator {
+            do {
+                let directoryAttributes = try directoryURL.resourceValues(forKeys: [.isDirectoryKey])
+                let dirExtension = directoryURL.pathExtension
+
+                if directoryAttributes.isDirectory! && resourceExtensions.contains(dirExtension) {
+                    let directoryEntry = directoryURL.relativePath.withoutPrefix(xcassetsRelativePath).withoutPrefix("/")
+                    let shouldBeKept = usedResourcesSet.contains(directoryEntry)
+
+                    if !shouldBeKept {
+                        try FileManager.default.removeItem(at: directoryURL)
+                    }
+
+                    print(" - \(directoryEntry) (\(shouldBeKept ? "kept" : "removed"))")
```

Comment:
while this is great should we print all the time tho?

Reply:
I think it'll be easier to correlate logs of our cloud builds to understand which assets were included in the bundle.
What is your main concern with printing all the time? Log size? Noise?

Reply:
log size and noise. can we put #ifdef?

Reply:
Okay. I'm leaving the verbose by default on the Debug configuration and suppressing it in the Release configuration. (attached screenshots of Release and Debug respectively:

**Release:**
![Release](https://user-images.githubusercontent.com/68076145/112874718-1eac0f80-9078-11eb-8ae1-0efc4975b6be.png)

**Debug:**
![Debug](https://user-images.githubusercontent.com/68076145/112874738-253a8700-9078-11eb-94dc-8861d79e082c.png)